docs/advanced/signal_generators.md

```diff
@@ -0,0 +1,558 @@
+# Signal Generation
+
+An overview of the signal generation in this package, along with information about Tektronix signal generators.
+
+______________________________________________________________________
+
+## Function Generation
+
+A function is a limited set of common waveforms that are provided by default through the instrument.
+The simplicity of these waveforms allows for output parameters like waveform length and sample rate to be
+abstracted away. Frequency replaces these in order to provide signals which easy to quantify and manipulate.
+
+Arbitrary Function Generators (AFGs) utilize a phase increment process and a data lookup table to provide variable frequencies.
+The phase increment calculated is dependent on the waveform length and frequency requested. This has
+a side effect where the phase increment can be larger than one index in the lookup table. Functions bypass this
+issue by being simplistic enough that waveform length reduction doesn't have a detrimental effect on the end output.
+
+Arbitrary Waveform Generators (AWGs) enforce one cycle per sample, allowing the output to be the same shape regardless of clock rate.
+With low frequency functions, AWGs are functionally identical to AFGs, besides offering more constrained amplitudes and offsets.
+
+______________________________________________________________________
+
+### Class Structure
+
+```{mermaid}
+classDiagram
+    direction LR
+
+    SignalGeneratorMixin <|-- Tekscope
+    SignalGeneratorMixin <|-- AFG
+    SignalGeneratorMixin <|-- AWG
+
+```
+
+The `SignalGenerator` class is responsible for most waveform generators, including the `AWG` and `AFG`.
+Similarly, `TekScope` is responsible for the AFG's internal to the scopes themselves, commonly referred to as
+an IAFG. All of these classes inherit `SignalGeneratorMixin`, which includes a list of methods which share
+functionality throughout all signal sources.
+
+```{note}
+SignalGeneratorMixin only contains abstract methods, defining the class by itself and calling methods in it will only raise NotImplemented errors.
```

Comment:
```suggestion
`SignalGeneratorMixin` only contains abstract methods; defining the class by itself and calling methods in it will only raise `NotImplementedError` errors.
```

I think the semi-colon works here.